app/models/manageiq/providers/redfish/physical_infra_manager.rb

```diff
@@ -25,6 +25,12 @@ class PhysicalInfraManager < ManageIQ::Providers::PhysicalInfraManager
     has_many :hardwares,
              :through => :physical_servers,
              :as      => :hardware
+    has_many :nics,
+             :through => :hardwares
+    has_many :storage_adapters,
+             :through => :hardwares
+    has_many :firmwares,
+             :through => :hardwares
```

Comment:
Same, can you add this to the core PhysicalInfraManager

Reply:
I took a quick look at Lenovo's [physical infra code](https://github.com/ManageIQ/manageiq-providers-lenovo/blob/master/app/models/manageiq/providers/lenovo/physical_infra_manager.rb). They already have some of this stuff present under different names. Should I copy their implementation into the core class, adjust the names in my PR, and create a new PR for Lenovo that removes the definitions that were added to core? Or is there another way you would like to see this resolved?

Reply:
I'd suggest using their names since the physical_server_hardwares is more general than just hardwares (also works with chassis storages switches), then adjust this PR to use those names and follow up with removing them from the lenovo repo.

Reply:
Just one last question. Should I copy the whole thing from Lenovo provider or just the definitions I need at the moment?

Reply:
Lets just copy all of them, all of the models for the other relationships (chassis, storage, switch) are in core so no reason not to have the associations in core also